.github/workflows/scripts/comments.mjs

```diff
@@ -1,14 +1,67 @@
+import { readFile } from 'node:fs/promises'
+
+/**
+ * Posts the content of a diff as a comment on a Github issue
+ *
+ * @param {GithubActionContext} githubActionContext
+ * @param {number} issueNumber
+ * @param {DiffComment} diffComment
+ */
+export async function commentDiff (
+  githubActionContext,
+  issueNumber,
+  { path, title, markerText }
+) {
+  // Read diff from previous step
+  const diffText = await readFile(path, 'utf8')
+
+  // Add or update comment on PR
+  try {
+    await comment(
+      githubActionContext,
+      issueNumber,
+      markerText,
+      renderDiffBodyText(title, diffText)
+    )
+  } catch {
+    await comment(
+      githubActionContext,
+      issueNumber,
+      markerText,
+      `## ${title}\n` +
+        `The diff could not be posted as a comment. You can download it from the [workflow artifacts](${githubActionRunUrl(
```

Comment:
I've kept this in addition to putting the link to the action run in the footer as:
- we want to replace any previous comment with something, so may as well be informative
- we want to point people to download the diffs, which are in the artifacts section of that page

Reply:
I found it a bit hard to follow but [added some suggestions](https://github.com/alphagov/govuk-frontend/pull/3671/files#r1202654624)

Reply:
If the `await comment()` crashes for another reason, "The diff could not be posted as a comment" is still true and it still got uploaded to the artifacts by the previous step in the workflow. That gives us a second chance at having a useful message posted on the PR (as well as to reserve an early comment if the posting fails because of an unlucky technical glitch on the first comment attempt).

I can try to get a large diff happening by changing the commit that triggers the changes in `dist` using say 4.4.1 or something for the JS.

Reply:
Sorry @romaricpascal I mostly meant we call `await comment()` from inside the catch block

So if the 2nd `await` rejects there's nothing to catch it and the build will unintentionally fail

Could always let it crash but with [`continue-on-error: true`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) and write a message?

1. [Adding a job summary](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary)
2. [Setting a warning message](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-warning-message)

![actions-job-summary-simple-example](https://github.com/alphagov/govuk-frontend/assets/415517/abb7e097-79b8-468e-ad33-23f377bc1466)

Reply:
But the backup comment is a bonus feature we can live without

Footer links are probably enough?

Would like to see how we handle crashes though (so we don't block a release PR merge)

Reply:
That's good

Maybe remove the try/catch and we can address it when it happens?

Reply:
Soo....

Turns out `continue-on-error` is a risky business, as it hides all errors from the workflow status (like [me forgetting to rename the function after introducting `commentDiffs`](https://github.com/alphagov/govuk-frontend/actions/runs/5082430875/jobs/9132138191?pr=3671)), so let's avoid it.

Simulating errors with `throw` in 2340312f5, this is the output we get, which is not too bad (only lists "Error" as a reason because I threw manually, not sure what would be logged for an actual errors from Github API), I think 😊 

```sh
Error: Posting diff comment failed
    at commentDiffs (file:///home/runner/work/govuk-frontend/govuk-frontend/.github/workflows/scripts/comments.mjs:26:11)
Error: Unhandled error: Error: Posting diff comment failed
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15143:16), <anonymous>:27:1)
    at async main (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15236:20) {
  [cause]: [
    {
      status: 'rejected',
      reason: Error
          at commentDiff (file:///home/runner/work/govuk-frontend/govuk-frontend/.github/workflows/scripts/comments.mjs:75:47)
          at processTicksAndRejections (node:internal/process/task_queues:96:5)
    }
  ]
}
```

With the workflow failed status, the trace in the workflow logs and the the difference in the hashes in the comments (if a comment was posted at all), I think that's plenty of information to start looking into things. We can add more logging if we investigate and realise there's more we need to speed us up the next times 😊